src/arr.ts

```diff
@@ -0,0 +1,196 @@
+					? [key, arrIds]
+					: undefined;
+			})
+			.filter((entry) => entry !== undefined) as [string, string][],
+	);
+}
+
+function formatFoundIds(arrJson: IdData): string {
+	let formattedIds = "";
+	keyNames.forEach((key) => {
+		if (arrJson[key]) {
+			formattedIds += `${chalk.yellow(
+				key.toUpperCase().replace("ID", ""),
+			)}: ${chalk.white(arrJson[key])} `;
+		}
+	});
+	return formattedIds.trim();
+}
+
+function logArrQueryResult(
+	arrJson: IdData,
+	searchTerm: string,
+	mediaType: MediaType,
+) {
+	const label = mediaType === MediaType.MOVIE ? Label.RADARR : Label.SONARR;
+	if (Object.keys(arrJson).length > 0) {
+		logger.verbose({
+			label: label,
+			message: `Found ${
+				label === Label.RADARR ? "movie" : "series"
+			} for ${chalk.green.bold(searchTerm)} -> ${formatFoundIds(
+				arrJson,
+			)}`,
+		});
+	} else {
+		logger.verbose({
+			label: label,
+			message: `Lookup failed for ${chalk.yellow(searchTerm)}`,
+		});
+		logger.verbose({
+			label: label,
+			message: `Make sure the ${
+				label === Label.RADARR ? "movie" : "series"
+			} is added to ${label === Label.RADARR ? "Radarr" : "Sonarr"}.`,
+		});
+	}
+}
+
+function logArrQueryFailure(error, searchTerm: string, mediaType: MediaType) {
+	const label = mediaType === MediaType.MOVIE ? Label.RADARR : Label.SONARR;
+	logger.debug({
+		label: label,
+		message: `Failed to lookup IDs for ${chalk.yellow(
+			searchTerm,
+		)} - (${chalk.red(String(error).split(":").slice(1)[0].trim())})`,
+	});
+	logger.debug(error);
+}
+
+function searchUArrLs(mediaType: MediaType): string[] | undefined {
+	const { sonarr, radarr } = getRuntimeConfig();
+	switch (mediaType) {
+		case MediaType.SEASON:
+		case MediaType.EPISODE:
+			return sonarr;
+		case MediaType.MOVIE:
+			return radarr;
+		default:
+			return undefined;
+	}
+}
+export async function grabArrId(
+	searchee: Searchee,
+	mediaType: MediaType,
+): Promise<Result<IdData, boolean>> {
+	const UArrLs = searchUArrLs(mediaType);
+	if (!UArrLs) {
+		return resultOfErr(false);
+	}
+	try {
+		let arrJson: IdData;
+		for (let i = 0; i < UArrLs.length; i++) {
+			arrJson = await fetchArrJSON(searchee, UArrLs[i]);
+			if (Object.keys(arrJson).length > 0) {
+				logArrQueryResult(arrJson, searchee.name, mediaType);
+				return resultOf(arrJson);
+			}
+		}
+		throw new Error("fall through to catch");
+	} catch (error) {
+		logArrQueryFailure(error, searchee.name, mediaType);
+		return resultOfErr(false);
+	}
+}
+export async function getRelevantArrIds(
+	searchee: Searchee,
+	ids: IdData,
+	caps: Caps,
+): Promise<IdSearchParams> {
+	const mediaType = getTag(searchee);
+	const idSearchCaps =
+		mediaType === MediaType.EPISODE || mediaType === MediaType.SEASON
+			? caps.tvIdSearch
+			: caps.movieIdSearch;
+
+	return {
+		tvdbid: idSearchCaps.tvdbId ? ids.tvdbId : undefined,
+		tmdbid: idSearchCaps.tvdbId ? ids.tmdbId : undefined,
+		imdbid: idSearchCaps.imdbId ? ids.imdbId : undefined,
+	};
```

Comment:
A bug was introduced with `getRelevantArrIds` where `shouldUseQuerySearch` is always false. This is due to the object always having all ids defined just their value being set to undefined. This causes searches without ids to not use the title.

Reply:
I mentioned this, and how it would behave, but it was late and IIRC it was time for bed. Thanks for fixing.